purchases/src/main/kotlin/com/revenuecat/purchases/common/events/FeatureEvent.kt

```diff
@@ -0,0 +1,28 @@
+package com.revenuecat.purchases.common.events
+
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.customercenter.events.CustomerCenterEvent
+import com.revenuecat.purchases.paywalls.events.PaywallEvent
+
+/**
+ * Feature related events in RevenueCat.
+ *
+ * **RevenueCatUI** features should define their events here.
+ */
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+sealed class FeatureEvent {
+
+    /**
+     * Represents an event related to paywalls.
+     *
+     * @property event The associated `PaywallEvent`.
+     */
+    data class Paywall(val event: PaywallEvent) : FeatureEvent()
```

Comment:
not a merge stopper but this could be directly the `PaywallEvent`? So you could make the `PaywallEvent` a `FeatureEvent`